packages/core/content-manager/server/controllers/collection-types.js

```diff
@@ -329,6 +329,50 @@ module.exports = {
 
     const number = await entityManager.getNumberOfDraftRelations(id, model);
 
+    return {
+      data: number,
+    };
+  },
+  async getMultipleEntriesNumberOfDraftRelations(ctx) {
+    const { userAbility } = ctx.state;
+    const { model, ids } = ctx.params;
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.read()) {
+      return ctx.forbidden();
+    }
+
+    const permissionQuery = await permissionChecker.sanitizedQuery.read(ctx.query);
+    const populate = await getService('populate-builder')(model)
+      .populateFromQuery(permissionQuery)
+      .build();
```

Comment:
I'm not really sure what this does, but I was thinking we would have set it up similar to:

https://github.com/strapi/strapi/blob/292e637aadf12c5e927be00dd1836e1b1cf90e1d/packages/core/content-manager/server/services/entity-manager.js#L378

🤔

Reply:
this has been recently added to populate fields for the permission-checker to see if user has permission to perform an action to some entitie(s).

With this a user can define custom conditions https://docs.strapi.io/user-docs/users-roles-permissions/configuring-administrator-roles#setting-custom-conditions-for-permissions to define if it can read/update/create entities.

In this case I am not completely sure about the use case and if it's necessary here, what is the use case to get the number of draft relations?

Nevertheless, I agree this sould be inside the entity-manager service and use the getDeepPopulateDraftCount

Reply:
Oh got it, thanks! I saw populate and assumed relations 🤦‍♂️ . I think we will need this as well. The use case is the similar to publishing single entity with relations in draft but for multiple entities:

Return the total number of relations in draft for each entity in order to warn the user they are about to publish an entity with relations in draft.

Reply:
I imagine in this case,  we should count the draft relations even if they don't have permissions to "see" those? If that were the case we might not need to check for read permissions.

Reply:
Yeah you're right that does make sense. I did notice that we have the check for the number of draft relations on a single entity:

https://github.com/strapi/strapi/blob/a56a6726fbfc09abca869929c1915f4c2bbb5cd9/packages/core/content-manager/server/controllers/collection-types.js#L428-L431

Maybe we should remove it there too?

Reply:
in the number of draft relations it mades sense in my opinion, I am in the process of creating the documentation for that bit of code, but in that case we only show users relations that has permissions to see, so we need this populate .